core/commands/dag/stat.go

```diff
@@ -3,73 +3,73 @@ package dagcmd
 import (
 	"fmt"
 	"io"
-	"os"
 
 	"github.com/ipfs/boxo/coreiface/path"
-	"github.com/ipfs/boxo/ipld/merkledag/traverse"
-	"github.com/ipfs/kubo/core/commands/cmdenv"
-	"github.com/ipfs/kubo/core/commands/e"
-
 	mdag "github.com/ipfs/boxo/ipld/merkledag"
+	cid "github.com/ipfs/go-cid"
 	cmds "github.com/ipfs/go-ipfs-cmds"
+	"github.com/ipfs/go-merkledag/traverse"
+	"github.com/ipfs/kubo/core/commands/cmdenv"
+	"github.com/ipfs/kubo/core/commands/e"
 )
 
 func dagStat(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
-	progressive := req.Options[progressOptionName].(bool)
-
 	api, err := cmdenv.GetApi(env, req)
 	if err != nil {
 		return err
 	}
-
-	rp, err := api.ResolvePath(req.Context, path.New(req.Arguments[0]))
-	if err != nil {
-		return err
-	}
-
-	if len(rp.Remainder()) > 0 {
-		return fmt.Errorf("cannot return size for anything other than a DAG with a root CID")
-	}
-
 	nodeGetter := mdag.NewSession(req.Context, api.Dag())
-	obj, err := nodeGetter.Get(req.Context, rp.Cid())
-	if err != nil {
-		return err
-	}
-
-	dagstats := &DagStat{}
-	err = traverse.Traverse(obj, traverse.Options{
-		DAG:   nodeGetter,
-		Order: traverse.DFSPre,
-		Func: func(current traverse.State) error {
-			dagstats.Size += uint64(len(current.Node.RawData()))
-			dagstats.NumBlocks++
 
-			if progressive {
-				if err := res.Emit(dagstats); err != nil {
-					return err
-				}
-			}
-			return nil
-		},
-		ErrFunc:        nil,
-		SkipDuplicates: true,
-	})
-	if err != nil {
-		return fmt.Errorf("error traversing DAG: %w", err)
-	}
+	cidSet := cid.NewSet()
+	dagStatSummary := &DagStatSummary{DagStatsArray: []*DagStat{}}
+	for _, a := range req.Arguments {
+		rp, err := api.ResolvePath(req.Context, path.New(a))
+		if err != nil {
+			return err
+		}
+		if len(rp.Remainder()) > 0 {
+			return fmt.Errorf("cannot return size for anything other than a DAG with a root CID")
+		}
 
-	if !progressive {
-		if err := res.Emit(dagstats); err != nil {
```

Comment:
AFAIT you fully removed support for the `progressive` option, we want to keep that, even if the information is partial (such as a simple total count and size sum equivalent to the previous option).

Reply:
I'm thinking of a way to output the progressive info for each CID and make it nicely formatted. Could it be like this? 

```
CID                                           	Blocks         	Size
QmcQs9AFNh4mHKqFLJ2Zo7WPSottWi9YDdEoqdk3jv5RyE	2200000        	20
QmcQs9AFNh4mHKqFLJ2Zo7WPSottWi9YDdEoqdk3jv5RyE	2200000        	20
QmSg4hRo947Lkp9qKn3KJhjjCFGNmtw8VLoXhUCC9mLRgU	2200000        	20
QmSg4hRo947Lkp9qKn3KJhjjCFGNmtw8VLoXhUCC9mLRgU	2200000        	20


Final Stats

CID                                           	Blocks         	Size
QmcQs9AFNh4mHKqFLJ2Zo7WPSottWi9YDdEoqdk3jv5RyE	2500000        	27
QmSg4hRo947Lkp9qKn3KJhjjCFGNmtw8VLoXhUCC9mLRgU	2500000        	27
QmSg4hRo947Lkp9qKn3KJhjjCFGNmtw8VLoXhUCC9mLRgU	2500000        	27

Summary

Total Size: 54
Unique Blocks: 2
Shared Size: 27
Ratio: 1.500000
```

I would make a tsv table with the progressive data and once it's done, Print a line indicating that it's the final stats and print the final table with the summary.

Reply:
Whatever you want, the current solution is extremely simple.

I basically want a way for users to see "something is happening". A simple counter that shows total blocks downloaded (like we have now) is fine.
If you want something more fancy feel free to do it but don't feel obligated to. :slightly_smiling_face:

Reply:
I'd rather focus on the other improvements you mentioned than doing something fancy for this right now. Can I keep the current solution with the same format for now?  It would probably look like this

```
Size: 21465226420, NumBlocks: 85379
Size: 21465226430, NumBlocks: 85389
Size: 21465226440, NumBlocks: 85399

Final Stats

CID                                           	Blocks         	Size
QmcQs9AFNh4mHKqFLJ2Zo7WPSottWi9YDdEoqdk3jv5RyE	2500000        	27
QmSg4hRo947Lkp9qKn3KJhjjCFGNmtw8VLoXhUCC9mLRgU	2500000        	27
QmSg4hRo947Lkp9qKn3KJhjjCFGNmtw8VLoXhUCC9mLRgU	2500000        	27

Summary

Total Size: 54
Unique Blocks: 2
Shared Size: 27
Ratio: 1.500000
```

Reply:
Look good to me.